hedera-node/src/main/java/com/hedera/services/keys/HederaKeyActivation.java

```diff
@@ -107,14 +113,30 @@ public static boolean otherPartySigsAreActive(
 			KeyActivationCharacteristics characteristics
 	) {
 		TransactionBody txn = accessor.getTxn();
-		Function<byte[], TransactionSignature> sigsFn = pkToSigMapFrom(accessor.getPlatformTxn().getSignatures());
+		Function<byte[], List<TransactionSignature>> sigsFn = pkToSigMapFrom(accessor.getPlatformTxn().getSignatures());
 
 		SigningOrderResult<SignatureStatus> othersResult = keyOrder.keysForOtherParties(txn, summaryFactory);
 		for (JKey otherKey : othersResult.getOrderedKeys()) {
-			if (!isActive(otherKey, sigsFn, ONLY_IF_SIG_IS_VALID, characteristics)) {
+			if (!isActive(otherKey, sigsFn, ONLY_IF_SIGS_ARE_VALID, characteristics)) {
 				return false;
 			}
 		}
+
+		if (accessor.getTxn().hasScheduleCreate() || accessor.getTxn().hasScheduleSign()) {
+			SignatureMap map;
+			if (accessor.getTxn().hasScheduleCreate()) {
+				map = accessor.getTxn().getScheduleCreate().getSigMap();
+			} else {
+				map = accessor.getTxn().getScheduleSign().getSigMap();
+			}
+
+			var keys = retrievePubKeys(map);
+			for (JKey key : keys) {
+				if (!isActive(key, sigsFn, ONLY_IF_SIGS_ARE_VALID, characteristics)) {
```

Comment:
This looks suspicious to me, for two reasons. One reason is small and the other perhaps larger...
1. _(Small)_ We don't require the `pubKeyPrefix` in a `SignaturePair` to be a full 32-byte Ed25519 public key; but only a unique prefix amongst all Ed25519 public keys referenced in the transaction's signing requirements. If we want to require the full 32-byte public key for sigMaps used in `ScheduleCreate/Sign`, we need to document that aggressively---and clear it with Leemon 😁 .

2. _(Larger?)_ The `List<JKey>` returned by `keyOrder.keysForOtherParties()` on L#118 contains the **Hedera keys** of entities referenced by the transaction (so could be simple Ed25519 keys, but also threshold keys, key lists, and even nested hierarchies of keys). We then use `isActive` to determine if enough Ed25519 keys signed to "activate" these complex key structures. But L#135 here makes no reference to the Hedera key structure, instead directly testing the Ed25519 keys found in the `sigMap` (and requiring them all to be valid). This wrong, since the validity of all Ed25519 signatures in the `sigMap` may well be **neither necessary nor sufficient** for the transaction's required Hedera keys to have active signatures.

My brain is a little fried today, will take a closer look tomorrow. 👍

Reply:
Regarding the second comment -> I think that we do not have a reason to worry because in this context we are verifying that the signatures provided in the `sigMap` on `create/sign` are signing the correct `transactionBody` and the signature is authentic (this is done first in the `signature expansion` where it is prepared, and then the result from the platform is validated here in this key activation context) 

Once the platform executes the verification and verifies that **the signatures in the sigmap are indeed signed and signed by the correctly specified pubkey**, the result is stored in `sigsFn` map. In the `otherPartySigsAreActive ` method we are simply verifying that all of the signatures from the `sigmap` are indeed signatures of the specified pubkey.

You are right that we should figure out the complex keys etc, but that is performed in the transition logic in order to determine:
- Whether the `JKey` is allowed to provide signature (is this JKey required based on `keysForOtherParties`)
- Whether all required `JKey`s provided signatures -> if they are, the transaction is ready for execution
but that's for another PR